src/modules/launcher/PowerLauncher/Plugin/PluginManager.cs

```diff
@@ -59,12 +58,12 @@ public static List<PluginPair> AllPlugins
                                 {
                                     try
                                     {
-                                        // Return a comparable produce version.
+                                        // Return a comparable product version.
                                         var fileVersion = FileVersionInfo.GetVersionInfo(x.ExecuteFilePath);
-                                        return ((uint)fileVersion.ProductMajorPart << 48)
-                                        | ((uint)fileVersion.ProductMinorPart << 32)
-                                        | ((uint)fileVersion.ProductBuildPart << 16)
-                                        | ((uint)fileVersion.ProductPrivatePart);
+                                        return ((uint)fileVersion.ProductMajorPart << 24)
+                                            | ((uint)fileVersion.ProductMinorPart << 16)
+                                            | ((uint)fileVersion.ProductBuildPart << 8)
+                                            | (uint)fileVersion.ProductPrivatePart;
```

Comment:
I think the idea here was to have it be long, most likely. How about doing the cast to `uint` first and then `ulong`?
```
return ((ulong)(uint)fileVersion.ProductMajorPart << 48)
    | ((ulong)(uint)fileVersion.ProductMinorPart << 32)
    | ((ulong)(uint)fileVersion.ProductBuildPart << 16)
    | (ulong)(uint)fileVersion.ProductPrivatePart;
```
perhaps with a comment that we're making it unsigned first and then extending to  64 bits.